pkg/driver/kubernetes/run.go

```diff
@@ -170,38 +170,49 @@ func (k *kubernetesDriver) runContainer(
 
 	// create the pod manifest
 	entrypoint, args := docker.GetContainerEntrypointAndArgs(mergedConfig, imageDetails)
-	pod := &corev1.Pod{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "Pod",
-			APIVersion: corev1.SchemeGroupVersion.String(),
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   id,
-			Labels: labels,
-		},
-		Spec: corev1.PodSpec{
-			ServiceAccountName: serviceAccount,
-			InitContainers:     initContainer,
-			Containers: []corev1.Container{
-				{
-					Name:         "devpod",
-					Image:        imageName,
-					Command:      []string{entrypoint},
-					Resources:    parseResources(k.config.Resources, k.Log),
-					Args:         args,
-					Env:          envVars,
-					VolumeMounts: volumeMounts,
-					SecurityContext: &corev1.SecurityContext{
-						Capabilities: capabilities,
-						Privileged:   mergedConfig.Privileged,
-						RunAsUser:    &[]int64{0}[0],
-						RunAsGroup:   &[]int64{0}[0],
-						RunAsNonRoot: &[]bool{false}[0],
-					},
-				},
-			},
-			RestartPolicy: corev1.RestartPolicyNever,
-			Volumes: []corev1.Volume{
+
+	var podTemplate *corev1.Pod
+	if len(k.config.PodManifestTemplate) > 0 {
+		podManifestTemplatePath := filepath.Join(mount.Source, k.config.PodManifestTemplate)
+		podTemplate, err = getPodTemplate(podManifestTemplatePath)
+		if err != nil {
+			return err
+		}
+	}
+
+	var pod *corev1.Pod
+	if podTemplate != nil {
+		pod = podTemplate
+		pod.ObjectMeta.Name = id
+		pod.ObjectMeta.Namespace = ""
+		pod.ObjectMeta.Labels = map[string]string{}
```

Comment:
We are resetting the pod Labels here, but I did not find any place where we are setting them back from the podTemplate and/or extraLabels from config. In such case, we may loose all labels or am I missing something here.

Reply:
fixed